sensor/kubernetes/fake/service.go

```diff
@@ -0,0 +1,170 @@
+			IPFamilies: []v1.IPFamily{v1.IPFamily(getIPFamily())},
+		},
+	}
+}
+
+func getNodePort(numLabels int) *v1.Service {
+	ns := namespacePool.mustGetRandomElem()
+	var labels map[string]string
+	if numLabels == 0 {
+		labels = createRandMap(16, 3)
+	} else {
+		labels = createMap(numLabels)
+	}
+	clusterIP := getRandIP()
+	return &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      randStringWithLength(16),
+			Namespace: ns,
+			UID:       newUUID(),
+		},
+		Spec: v1.ServiceSpec{
+			Type:     v1.ServiceTypeNodePort,
+			Selector: labels,
+			Ports: []v1.ServicePort{
+				{
+					Protocol: v1.Protocol(getRandProtocol()),
+					Port:     int32(getRandPort()),
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: int32(getRandPort()),
+					},
+					NodePort: int32(getRandPort()),
+				},
+			},
+			ClusterIP:  clusterIP,
+			ClusterIPs: []string{clusterIP},
+			IPFamilies: []v1.IPFamily{v1.IPFamily(getIPFamily())},
+		},
+	}
+}
+
+func getLoadBalancer(numLabels int) *v1.Service {
+	ns := namespacePool.mustGetRandomElem()
+	var labels map[string]string
+	if numLabels == 0 {
+		labels = createRandMap(16, 3)
+	} else {
+		labels = createMap(numLabels)
+	}
+	clusterIP := getRandIP()
+	internalTrafficPolicy := v1.ServiceInternalTrafficPolicyCluster
+	allocateLoadBalancerNodePorts := true
+	ipFamilyPolicy := v1.IPFamilyPolicySingleStack
+	return &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      randStringWithLength(16),
+			Namespace: ns,
+			UID:       newUUID(),
+		},
+		Spec: v1.ServiceSpec{
+			Type:     v1.ServiceTypeLoadBalancer,
+			Selector: labels,
+			Ports: []v1.ServicePort{
+				{
+					Protocol: v1.Protocol(getRandProtocol()),
+					Port:     int32(getRandPort()),
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: int32(getRandPort()),
+					},
+					NodePort: int32(getRandPort()),
+				},
+			},
+			ClusterIP:                     clusterIP,
+			ClusterIPs:                    []string{clusterIP},
+			IPFamilies:                    []v1.IPFamily{v1.IPFamily(getIPFamily())},
+			ExternalTrafficPolicy:         v1.ServiceExternalTrafficPolicyTypeCluster,
+			InternalTrafficPolicy:         &internalTrafficPolicy,
+			AllocateLoadBalancerNodePorts: &allocateLoadBalancerNodePorts,
+			PublishNotReadyAddresses:      false,
+			IPFamilyPolicy:                &ipFamilyPolicy,
+		},
+		Status: v1.ServiceStatus{
+			LoadBalancer: v1.LoadBalancerStatus{
+				Ingress: []v1.LoadBalancerIngress{
+					{
+						IP: getRandIP(),
+					},
+				},
+			},
+		},
+	}
+}
+
+func getService(workload ServiceWorkload) []runtime.Object {
+	objects := make([]runtime.Object, 0, workload.NumClusterIPs+workload.NumNodePorts+workload.NumLoadBalancers)
+	for i := 0; i < workload.NumClusterIPs; i++ {
+		clusterIP := getClusterIP(workload.NumLabels)
+		objects = append(objects, clusterIP)
+	}
+	for i := 0; i < workload.NumNodePorts; i++ {
+		clusterIP := getNodePort(workload.NumLabels)
```

Comment:
Probably due to copy-paste
```suggestion
		numPort := getNodePort(workload.NumLabels)
```

Reply:
Fixed